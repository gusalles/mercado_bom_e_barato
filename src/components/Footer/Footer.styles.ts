import { Divider } from '@mui/material';
import { styled } from '@mui/material/styles';

export const FooterContainer = styled('footer')(({ theme }) => ({
  width: '100%',
  padding: '60px 96px',
  backgroundColor: theme.palette.primary.main,
  display: 'block',
  [theme.breakpoints.down('md')]: {
    display: 'none',
  },
}));

export const FooterSection = styled('section')(() => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
}));

export const FooterDivider = styled(Divider)(() => ({
  marginTop: '32px',
  marginBottom: '16px',
}));

export const FooterUseTerm = styled('p')(() => ({
  color: 'black',
  fontSize: '0.75rem',
  lineHeight: '160%',
}));
