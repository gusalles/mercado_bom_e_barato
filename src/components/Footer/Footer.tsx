import { Box } from '@mui/material';

export function Footer() {
  return (
    <Box
      component="footer"
      sx={{
        position: 'fixed',
        bottom: 0,
        width: '100%',
        textAlign: 'center',
        padding: '1rem',
        backgroundColor: '#f8f8f8',
      }}
    >
      Ice Dreams ©️ 2024 • Política de Privacidade • Termos de Uso
    </Box>
  );
}
