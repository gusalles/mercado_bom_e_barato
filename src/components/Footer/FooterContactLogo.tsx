import Image from 'next/image';

interface FooterContactLogoProps {
  url: string;
  alt: string;
}

export function FooterContactLogo({ url, alt }: FooterContactLogoProps) {
  return <Image src={url} alt={alt} width={24} height={24} />;
}
