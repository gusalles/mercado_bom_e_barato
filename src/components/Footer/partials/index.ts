export * from './FooterContact';
export * from './FooterContactLogo';
