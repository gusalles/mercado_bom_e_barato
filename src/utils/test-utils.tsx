import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
