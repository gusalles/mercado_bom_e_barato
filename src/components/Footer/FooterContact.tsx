import { FooterContactLogo } from './FooterContactLogo';

export function FooterContact() {
  return (
    <div
      style={{
        display: 'flex',
        alignItems: 'center',
        gap: '1rem',
        color: 'black',
        fontWeight: 600,
        fontSize: '1rem',
        lineHeight: '140%',
      }}
    >
      Acompanhe nossas redes sociais:
      <FooterContactLogo url="/youtube.svg" alt="YouTube" />
      <FooterContactLogo url="/spotify.svg" alt="Spotify" />
      <FooterContactLogo url="/facebook.svg" alt="Facebook" />
      <FooterContactLogo url="/instagram.svg" alt="Instagram" />
    </div>
  );
}
