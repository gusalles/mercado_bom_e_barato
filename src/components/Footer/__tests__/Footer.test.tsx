import { render, screen } from '@/utils/test-utils';

import { Footer } from '../Footer';

describe('Footer', () => {
  it('should render', () => {
    render(<Footer />);

    const footer = screen.getByRole('contentinfo');

    expect(footer).toMatchSnapshot();
  });
});
