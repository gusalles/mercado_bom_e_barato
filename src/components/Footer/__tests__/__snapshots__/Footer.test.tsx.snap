// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should render 1`] = `
<footer
  class="css-gqnc24"
>
  <section
    class="css-69i1ev"
  >
    <img
      alt="Logo Mercado B&B"
      data-nimg="1"
      decoding="async"
      height="40"
      src="/logo.svg"
      style="color: transparent;"
      width="145"
    />
    <div
      class="css-1bt2vw6"
    >
      Acompanhe nossas redes sociais:
      <img
        alt="YouTube"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/youtube.svg"
        style="color: transparent;"
        width="24"
      />
      <img
        alt="Spotify"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/spotify.svg"
        style="color: transparent;"
        width="24"
      />
      <img
        alt="Facebook"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/facebook.svg"
        style="color: transparent;"
        width="24"
      />
      <img
        alt="Instagram"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/instagram.svg"
        style="color: transparent;"
        width="24"
      />
    </div>
  </section>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1c1tr4e-MuiDivider-root"
  />
  <section
    class="css-69i1ev"
  >
    <p
      class="css-12cepmv"
    >
      Ice Dreams ©️ 2024 • Política de Privacidade • Termos de Uso
    </p>
  </section>
</footer>
`;
